### PR TITLE
[SPIR-V] Enable more unsupported tests

### PR DESCRIPTION
**Pull request by sudonatalie**
The following tests only required minor test syntax changes to pass:
- tools/clang/test/CodeGenSPIRV/cast.2float.interlocked.hlsl
- tools/clang/test/CodeGenSPIRV/meshshading.nv.error.fncall.amplification.vulkan1.2.hlsl (+ replacing NV ext with EXT)
- tools/clang/test/CodeGenSPIRV/var.init.extvector.hlsl

Issue #6621 has been filed to track the failure of tools/clang/test/CodeGenSPIRV/oo.class.static.member.hlsl.

Related to #6616

### CODE REVIEW

**Review by Keenuts — APPROVED** [review]

File: tools/clang/test/CodeGenSPIRV/meshshading.nv.error.fncall.amplification.vulkan1.2.hlsl [hunk 1]
```diff
@@ -1,8 +1,8 @@
-// RUN: %dxc -T as_6_5 -E main -fspv-target-env=vulkan1.2
+// RUN: %dxc -T as_6_5 -E main -fspv-target-env=vulkan1.2 -fcgl -spirv %s | FileCheck %s
 
-// CHECK:  OpCapability MeshShadingNV
-// CHECK:  OpExtension "SPV_NV_mesh_shader"
-// CHECK:  OpEntryPoint TaskNV %main "main"
```
Comment (s-perron) [inline tools/clang/test/CodeGenSPIRV/meshshading.nv.error.fncall.amplification.vulkan1.2.hlsl:5]: Why did you remove the "CHECK"?

**Review by s-perron — APPROVED** [review]
Approved except for a potential missing "CHECK".

File: tools/clang/test/CodeGenSPIRV/meshshading.nv.error.fncall.amplification.vulkan1.2.hlsl (hunk 1 above)
Comment (sudonatalie) [reply tools/clang/test/CodeGenSPIRV/meshshading.nv.error.fncall.amplification.vulkan1.2.hlsl:5]: Oops, totally unintentional. Good catch.
